sc-memory/sc-memory/sc_memory.hpp

```diff
@@ -87,61 +120,407 @@ class ScMemoryContext
+   *
+   * @code
+   * ScMemoryContext ctx;
+   * ScAddr nodeAddr = ctx.CreateNode(ScType::NodeConst);
+   * @endcode
+   */
   _SC_EXTERN ScAddr CreateNode(ScType const & type);
+
+  /*!
+   * @brief Creates a new sc-link with the specified type.
+   *
+   * This method creates a new sc-link with the specified type and returns its sc-address.
+   *
+   * @param type The type of the sc-link to create (default is ScType::LinkConst).
+   * @return Returns the sc-address of the newly created sc-link.
+   * @throws ExceptionInvalidParams if the specified type is not a valid sc-link type or if sc-memory is full.
+   *
+   * @example
+   * @code
+   * ScMemoryContext ctx;
+   * ScAddr linkAddr = ctx.CreateLink();
+   * @endcode
+   */
   _SC_EXTERN ScAddr CreateLink(ScType const & type = ScType::LinkConst);
 
+  /*!
+   * @brief Creates a new sc-connector with the specified type, source, and target.
+   *
+   * This method creates a new sc-connector with the specified type, source, and target, and returns its sc-address.
+   *
+   * @param type The type of the sc-connector to create.
+   * @param addrBeg The sc-address of the source sc-element.
+   * @param addrEnd The sc-address of the target sc-element.
+   * @return Returns the sc-address of the newly created sc-connector.
+   * @throws ExceptionInvalidParams if the specified source or target sc-address is invalid or if sc-memory is full.
+   *
+   * @example
+   * @code
+   * ScMemoryContext ctx;
+   * ScAddr sourceNode = ctx.CreateNode(ScType::NodeConst);
+   * ScAddr targetNode = ctx.CreateNode(ScType::NodeConst);
+   * ScAddr edgeAddr = ctx.CreateEdge(ScType::EdgeDCommonConst, sourceNode, targetNode);
+   * @endcode
+   */
   _SC_EXTERN ScAddr CreateEdge(ScType const & type, ScAddr const & addrBeg, ScAddr const & addrEnd);
 
-  //! Returns type of sc-element. If there are any error, then returns ScType::Unknown
+  /*!
+   * @brief Returns the type of the specified sc-element.
+   *
+   * This method retrieves the type of the sc-element identified by the given sc-address.
+   *
+   * @param addr The sc-address of the sc-element to query.
+   * @return Returns the type of the specified sc-element.
+   * @throws ExceptionInvalidParams if the specified sc-address is invalid.
+   *
+   * @example
+   * @code
+   * ScMemoryContext ctx;
+   * ScAddr elementAddr = ctx.CreateNode(ScType::NodeConst);
+   * ScType elementType = ctx.GetElementType(elementAddr);
+   * std::cout << "Element Type: " << elementType.GetName() << std::endl;
+   * @endcode
+   */
   _SC_EXTERN ScType GetElementType(ScAddr const & addr) const;
 
-  /*! Change subtype of sc-element.
-   * Return true, if there are no any errors; otherwise return false.
+  /*!
+   * @brief Changes the subtype of an sc-element.
+   *
+   * This method changes the subtype of the sc-element identified by the given sc-address.
+   *
+   * @param addr The sc-address of the sc-element to modify.
+   * @param subtype The new subtype to set.
```

Comment:
Why not ScType? A sc_type

Reply:
I fix it in another PR. where I must deprecated and clarify all C++ API methods.